src/hooks/http.js

```diff
@@ -68,14 +68,35 @@ export const parseHttpRequestOptions = (options = {}, url) => {
     port,
     uri,
     host,
-    body: '', // XXX Filled by the httpRequestEndWrapper ( / Write)
+    body: '', // XXX Filled by the httpRequestEndWrapper or httpRequestEmitWrapper ( / Write)
     method,
     headers: lowerCaseObjectKeys(headers),
     protocol,
     sendTime,
   };
 };
 
+export const httpRequestEmitWrapper = requestData => emitFunc =>
+  function(eventName, arg) {
+    safeExecute(
+      () => {
+        if (arg && arg._httpMessage && arg._httpMessage._hasBody) {
+          const httpMessage = arg._httpMessage;
+          let lines = [];
+          if (httpMessage.hasOwnProperty('outputData')) {
+            lines = httpMessage.outputData[0].data.split('\n');
+          } else if (httpMessage.hasOwnProperty('output')) {
+            lines = httpMessage.output[0].split('\n');
+          }
+          requestData.body += lines[lines.length - 1];
```

Comment:
why do you take only the last line?
Don't you want to take all the lines to the body?

Reply:
Moreover, if lines is empty, then you mess `requestData.body` with `+=undefined` which cast it to string 😂

Reply:
Fixed :)